sdk/cosmos/azure_data_cosmos/src/clients/container_client.rs

```diff
@@ -0,0 +1,230 @@
+    /// * `options` - Optional parameters for the request.
+    ///
+    /// # Examples
+    ///
+    /// The `query` and `partition_key` parameters accept anything that can be transformed [`Into`] their relevant types.
+    /// This allows simple queries without parameters to be expressed easily:
+    ///
+    /// ```rust,no_run
+    /// # async fn doc() {
+    /// # use azure_data_cosmos::{CosmosClient, CosmosClientMethods, clients::DatabaseClientMethods, clients::ContainerClientMethods};
+    /// # let credential = azure_identity::create_default_credential().unwrap();
+    /// # let client = CosmosClient::new("https://myaccount.documents.azure.com/", credential, None).unwrap();
+    /// # let db_client = client.database_client("my_database");
+    /// # let container_client = db_client.container_client("my_container");
+    /// #[derive(serde::Deserialize)]
+    /// struct Customer {
+    ///     id: u64,
+    ///     name: String,
+    /// }
+    /// let items = container_client.query_items::<Customer>(
+    ///     "SELECT * FROM c",
+    ///     "some_partition_key",
+    ///     None).unwrap();
+    /// # }
+    /// ```
+    ///
+    /// You can specify parameters by using [`Query::from()`] and [`Query::with_parameter()`]:
+    ///
+    /// ```rust,no_run
+    /// # async fn doc() {
+    /// # use azure_data_cosmos::{CosmosClient, CosmosClientMethods, clients::DatabaseClientMethods, clients::ContainerClientMethods, Query};
+    /// # let credential = azure_identity::create_default_credential().unwrap();
+    /// # let client = CosmosClient::new("https://myaccount.documents.azure.com/", credential, None).unwrap();
+    /// # let db_client = client.database_client("my_database");
+    /// # let container_client = db_client.container_client("my_container");
+    /// #[derive(serde::Deserialize)]
+    /// struct Customer {
+    ///     id: u64,
+    ///     name: String,
+    /// }
+    /// let query = Query::from("SELECT COUNT(*) FROM c WHERE c.customer_id = @customer_id")
+    ///     .with_parameter("@customer_id", 42).unwrap();
+    /// let items = container_client.query_items::<Customer>(query, "some_partition_key", None).unwrap();
+    /// # }
+    /// ```
+    ///
+    /// See [`PartitionKey`] for more information on how to specify a partition key, and [`Query`] for more information on how to specify a query.
+    fn query_items<T: DeserializeOwned + Send>(
+        &self,
+        query: impl Into<Query>,
+        partition_key: impl Into<PartitionKey>,
+        options: Option<QueryOptions>,
+    ) -> azure_core::Result<azure_core::Pageable<QueryResults<T>, azure_core::Error>>;
+}
+
+/// A client for working with a specific container in a Cosmos DB account.
+///
+/// You can get a `Container` by calling [`DatabaseClient::container_client()`](crate::clients::DatabaseClient::container_client()).
+pub struct ContainerClient {
+    base_url: Url,
+    root_client: CosmosClient,
```

Comment:
This might end up being more expensive. Why not just clone the pipeline, which is cheap? If you have strings on the client e.g., the `endpoint`, you presumably just cloned that as well.